### PR TITLE
feat(web-analytics): Add columns to loading state, dedupe some code

### PR DESCRIPTION
**Pull request by robbie-c**
## Problem

The loading state of web analytics was pretty rough before this. See https://posthog.slack.com/archives/C05LJK1N3CP/p1725486074884769

## Changes
Show column names as part of loading state:
Screenshot 2024-09-12 at 11 32 57

A lot of the query creation logic is simplified with some helper functions

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Mostly through staring a lot at the dashboard

### CODE REVIEW

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx [hunk 1]
```diff
@@ -254,27 +259,14 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
...
-            state,
-        }),
         openModal: (tileId: TileId, tabId?: string) => {
             return { tileId, tabId }
         },
         closeModal: () => ({}),
         openAsNewInsight: (tileId: TileId, tabId?: string) => {
             return { tileId, tabId }
         },
+        setHasLoadedStateFromUrl: () => ({}),
```
Comment (daibhin) [inline frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:269]:
```suggestion
        setHasLoadedStateFromUrl: true,
```

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx [hunk 2]
```diff
@@ -1714,6 +1401,11 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
                 filter_test_accounts,
             }
         ) => {
+            if (values.hasLoadedStateFromUrl) {
+                // only load from URL once, to prevent infinite loops
+                return
+            }
```
Comment (daibhin) [inline frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:1407]: It feels wrong having to do this 🤔

Looking through the codebase it doesn't seem to be the case that we have to do this elsewhere. `urlToAction` to seems to protect itself somehow, but now that I think about it I'm not sure how

**Review by daibhin — APPROVED** [review]
Not really a blocker but one thing feels off

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx (hunk 1 above)
Comment (robbie-c) [reply frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:269]: Hmm I tried this and it complained, might have just been that the types hadn't caught up yet, so I'll try this again

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx (hunk 2 above)
Comment (robbie-c) [reply frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:1407]: I don't need it - we don't get infinite loops without it, maybe I'm just being paranoid

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx (hunk 1 above)
Comment (robbie-c) [reply frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx]: Done

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx (hunk 2 above)
Comment (robbie-c) [reply frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx]: I've removed it

**Pull request merged**